workers/data_refinery_workers/processors/smasher.py

```diff
@@ -339,11 +410,18 @@ def _notify(job_context: Dict) -> Dict:
             SENDER = "Refine.bio Mail Robot <noreply@refine.bio>"
             RECIPIENT = job_context["dataset"].email_address
             AWS_REGION = "us-east-1"
-            SUBJECT = "Your refine.bio Dataset is Ready!"
-            BODY_TEXT = "Hot off the presses:\n\n" + job_context["result_url"] + "\n\nLove!,\nThe refine.bio Team"
-            BODY_HTML = "Hot off the presses:<br /><br />" + job_context["result_url"] + "<br /><br />Love!,<br />The refine.bio Team"
             CHARSET = "UTF-8"
 
+            if job_context['job'].failure_reason not in ['', None]:
+                SUBJECT = "Your refine.bio Dataset is Ready!"
+                BODY_TEXT = "Hot off the presses:\n\n" + job_context["result_url"] + "\n\nLove!,\nThe refine.bio Team"
+                BODY_HTML = "Hot off the presses:<br /><br />" + job_context["result_url"] + "<br /><br />Love!,<br />The refine.bio Team"
+            else:
+                SUBJECT = "There was a problem processing your refine.bio dataset :("
+                BODY_TEXT = "We tried but were unable to process your requested dataset. Error was: \n\n" + str(job_context['job'].failure_reason) + "\nDataset ID: " + str(dataset.id) + "\n We have been notified and are looking into the problem. \n\nSorry!"
+                BODY_HTML = BODY_TEXT = "We tried but were unable to process your requested dataset. Error was: <br /><br />" + job_context['job'].failure_reason + "<br />Dataset: " + str(dataset.id) + "<br /> We have been notified and are looking into the problem. <br /><br />Sorry!"
+                job_context['success'] = False
```

Comment:
Since you're doing this regardless of whether this is cloud or not below, does this really need to be here?

Reply:
There is a small window where it hasn't been sent yet and it could be interruped. It shouldn't really matter but I don't want to re-do a job if the email is sent. I don't think it matters too much.